packages/core/src/browser/menu/browser-menu-plugin.ts

```diff
@@ -468,9 +468,11 @@ export class MenuCommandRegistry extends PhosphorCommandRegistry {
         });
 
         const bindings = keybindingRegistry.getKeybindingsForCommand(id);
-        // Only consider the first keybinding.
+        // Only consider the first active keybinding.
         if (bindings.length) {
-            const binding = bindings[0];
+            const binding = bindings.length > 1 ?
+                bindings.find(b => !b.when || this.services.contextKeyService.match(b.when)) ?? bindings[0] :
+                bindings[0];
```

Comment:
Suggestion: Do we really need the conditional? Given that we always fall back to `bindings[0]` anyway?
```suggestion
            const binding = bindings.find(b => !b.when || this.services.contextKeyService.match(b.when)) ?? bindings[0]
```

Reply:
@msujew it is a small performance improvement. Most keybindings only have one binding. with the conditional it does not have to evaluate the when clause for every single keybinding

Reply:
Ok, I can see that 👍